part1.js

```diff
@@ -5,9 +5,40 @@
    dimensions of a box to be wrapped, for example:
      ['2x3x4', '3x5x8', '1x1x10']
 */
-function paperCalculator(boxes) {
-  // TODO: return the correct amount of wrapping paper needed.
-  return 0;
-};
 
-exports.paperCalculator = paperCalculator;
+var boxSamples = ['2x3x4', '3x5x8', '1x1x10'];
+
+function paperCalculator(arrOfBoxes) {
+  var total = 0;
+  for (i = 0; i < arrOfBoxes.length; i++) {
+    total += paperCalculatorHelper(arrOfBoxes[i]);
+  }
+  return total;
+}
+
+function paperCalculatorHelper(box) {
+
+  var l = box.slice(0, box.indexOf('x'));
+  var w = box.slice((box.indexOf('x') + 1), box.lastIndexOf('x'));
+  var h = box.slice(box.lastIndexOf('x') + 1);
+  var lInt = parseInt(l, 10);
+  var wInt = parseInt(w, 10);
+  var hInt = parseInt(h, 10);
+
+  var sizes = [(lInt * wInt), (wInt * hInt), hInt * lInt];
+
+  sizes.sort(function(a, b) {
+    return a - b;
+  });
```

Comment:
Since this is the default behavior for sort, you don't need to pass in a custom compare function:
```javascript
sizes.sort();
```